system/jlib/jtrace.cpp

```diff
@@ -107,6 +108,260 @@ class NoopSpanExporterFactory
+            return opentelemetry::sdk::common::ExportResult::kFailure;
+        }
+
+        for (auto &recordable : recordables)
+        {
+            auto span = std::unique_ptr<opentelemetry::sdk::trace::SpanData>(
+            static_cast<opentelemetry::sdk::trace::SpanData *>(recordable.release()));
+
+            if (span != nullptr)
+            {
+                char traceID[32]       = {0};
+                char spanID[16]        = {0};
+                
+
+                span->GetTraceId().ToLowerBase16(traceID);
+                span->GetSpanId().ToLowerBase16(spanID);
+                
+
+                StringBuffer out;
+                out.setf("{ \"Name\": \"%s\"", span->GetName().data());
+                out.append(", \"TraceId\": \"").append(32, traceID).append("\"");
+                out.append(", \"SpanId\": \"").append(16, spanID).append("\"");
+                out.appendf(", \"Start\": %ld", span->GetStartTime().time_since_epoch().count());
+                out.appendf(", \"Duration\": %ld", span->GetDuration().count());
+
+                if (hasMask(logFlags, SpanLogFlags::LogParentInfo))
+                {
+                    if (span->GetParentSpanId().IsValid())
+                    {
+                        char parentSpanID[16]  = {0};
+                        span->GetParentSpanId().ToLowerBase16(parentSpanID);
+                        out.append(", \"ParentSpanId\": \"").append(16, parentSpanID).append("\"");
+                    }
+
+                    const char * traceState = span->GetSpanContext().trace_state()->ToHeader().c_str();
+                    if (!isEmptyString(traceState))
+                        out.appendf(", \"TraceState\": \"%s\"", traceState);
+                }
+
+                if (hasMask(logFlags, SpanLogFlags::LogSpanDetails))
+                {
+                    out.appendf(", \"Status\": \"%s\"", statusMap[int(span->GetStatus())].c_str());
+                    out.appendf(", \"kind\": \"%s\"", spanKindToString(span->GetSpanKind()));
+                    const char * description = span->GetDescription().data();
+                    if (!isEmptyString(description))
+                        out.appendf(", \"Description\": \"%s\"", description);
+                    printInstrumentationScope(out, span->GetInstrumentationScope());
+                }
+
+                if (hasMask(logFlags, SpanLogFlags::LogAttributes))
+                    printAttributes(out, span->GetAttributes());
+
+                if (hasMask(logFlags, SpanLogFlags::LogEvents))
+                    printEvents(out, span->GetEvents());
+
+                if (hasMask(logFlags, SpanLogFlags::LogLinks))
+                    printLinks(out, span->GetLinks());
+
+                if (hasMask(logFlags, SpanLogFlags::LogResources))
+                    printResources(out, span->GetResource());
+
+                out.append(" }");
+                LOG(MCmonitorEvent, "%s",out.str());
```

Comment:
As it stands there is no indication that this is a span - and I this will not be the only MON EVT.

Adding `"event": "SpanFinish",` to the json would work (although it is more verbose).

Reply:
Ok, so clearly you have something else in mind that could be "MON EVT", but I'm not sure if that was made public anywhere.
Adding an inline json entry won't help scanners/scrappers efficiently detect spans. It would be more effective to explicitly label the log type as a span. Without the benefit of a log class "span" your original text prefix is prob the next best option

Reply:
Opting for inline type: span